sampi/rpc.go

```
@@ -93,3 +95,29 @@ func (s *Cluster) RemoveJobs(hostID *string, jobIDs []string, res *struct{}) err
 	s.state.Commit()
 	return nil
 }
+
+func (s *Cluster) StreamHostEvents(id string, stream rpcplus.Stream) error {
+	ch := make(chan host.HostEvent)
+	s.state.AddListener(id, ch)
+	defer func() {
+		go func() {
+			// drain to prevent deadlock while removing the listener
+			for _ = range ch {
+			}
+		}()
+		s.state.RemoveListener(id, ch)
+		close(ch)
+	}()
+	for {
+		select {
+		case event := <-ch:
+			select {
+			case stream.Send <- event:
+			case <-stream.Error:
+				return nil
```

You should either return the error from `stream.Error`, or drop the return value of this method and just `break` here